modules/blocks/index.js

```diff
@@ -79,3 +85,33 @@ export function getBlockSettings( slug ) {
 export function getBlocks() {
 	return Object.values( blocks );
 }
+
+/**
+ * Returns the element of a registered block node given a context and its
+ * parsed metadata.
+ *
+ * @param  {Object}     blockNode Parsed block node
+ * @param  {String}     context   Render context ("edit", "save")
+ * @return {?WPElement}           Block element, or undefined if type unknown
+ */
+export function createBlockElement( blockNode, context = 'edit' ) {
+	const { blockType, rawContent } = blockNode;
+
+	// Verify block is of known type
+	const block = getBlockSettings( blockType );
+	if ( ! block ) {
+		return;
+	}
+
+	// Merge attributes from parse with block implementation
+	let { attrs } = blockNode;
+	if ( 'function' === typeof block.attributes ) {
+		attrs = { ...attrs, ...block.attributes( rawContent ) };
+	} else if ( block.attributes ) {
+		attrs = { ...attrs, ...query.parse( rawContent, block.attributes ) };
+	}
+
+	if ( 'function' === typeof block[ context ] ) {
+		return block[ context ]( attrs );
+	}
+}
```

Comment:
>I feel this method should live under a `parser` folder, What do you think of a folder structure like this: `parser/grammar.js`,  `parser/index.js` and the parser in `parser/index.js` calls the grammar and then map on the blocks and return directly the final "WPBlocks".

This could make sense, only because it's value alone assumes a caller knows the `blockNode` parsed object structure, which... is not likely. Maybe better to tie in with base parsing logic like you mentioned.